SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -349,34 +362,43 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 
 			if (SpatialGDKSettings->bEnableClientResultTypes)
 			{
-				NewQuery.ResultComponentId = ClientResultType;
+				NewQuery.ResultComponentId = ClientNonAuthInterestResultType;
 			}
 			else
 			{
 				NewQuery.FullSnapshotResult = true;
 			}
 
-			ClientComponentInterest.Queries.Add(NewQuery);
+			AddComponentQueryPairToInterestComponent(OutInterest, ClientEndpointComponentId, NewQuery);
 		}
 	}
+}
 
-	Interest NewInterest;
-	// Server Interest
-	if (SystemConstraints.IsValid() && SpatialGDKSettings->bEnableServerQBI)
-	{
-		NewInterest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, ServerComponentInterest);
-	}
-	// Client Interest
-	if (ClientConstraint.IsValid())
+void InterestFactory::AddClientSelfInterest(Interest& OutInterest, const Worker_EntityId EntityId) const
+{
+	Query NewQuery;
+	// Just an entity ID constraint is fine, as clients should not become authoritative over entities outside their loaded levels
+	NewQuery.Constraint.EntityIdConstraint = EntityId;
+
+	NewQuery.ResultComponentId = ClientAuthInterestResultType;
+
+	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::GetClientAuthorityComponent(GetDefault<USpatialGDKSettings>()->bUseRPCRingBuffers), NewQuery);
+}
+
+void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& OutInterest, const Worker_ComponentId ComponentId, const Query& QueryToAdd)
+{
+	if (!OutInterest.ComponentInterestMap.Contains(ComponentId))
 	{
-		NewInterest.ComponentInterestMap.Add(SpatialConstants::GetClientAuthorityComponent(SpatialGDKSettings->bUseRPCRingBuffers), ClientComponentInterest);
+		ComponentInterest NewComponentInterest;
+		OutInterest.ComponentInterestMap.Add(ComponentId, NewComponentInterest);
 	}
-
-	return NewInterest;
+	auto& Queries = OutInterest.ComponentInterestMap[ComponentId].Queries;
+	Queries.Add(QueryToAdd);
 }
 
-void InterestFactory::AddActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries, bool bRecurseChildren) const
+void InterestFactory::GetActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries, bool bRecurseChildren) const
 {
+	TArray<Query> Queries;
```

Comment:
What do we do with these queries? Should they be added to OutQueries?

Reply:
That's a bug, made a mistake when resolving merge conflicts. Fixed